### DISCUSSION TITLE
Upgrading hydra from 1.1.0 to  1.10.2 apis sometimes takes more then 15 seconds

### DISCUSSION THREAD

**Discussion by azajkhan** [ory/hydra#2867] [Q&A] [answered]
I am using ORY hydra 1.10.2,
Recently while checking logs we observed sometimes ORY hydra APIs takes more than 15 sec to serve the request.
Its very intermittent, sometimes same api give response in milliseconds.
We have a dedicated server running  ORY hydra 1.10.2 and there are not many concurrent requests on this env.
Can someone help me understand , why hydra sometimes takes 15-20 (or more) sec to serve request, and immediately next moment same apis responds instantly.
This is observed on hydra upgrade from 1.1.0 to 1.10.2.
We did not change anything on BCrypt factor on side.
We started getting this all of a sudden on an env where not much load is there. It is a controlled env with some limited users.

We did see a few slow queries for the below 2 inserts: these seem to take a couple of seconds -> 5+ seconds.
INSERT INTO hydra_oauth2_authentication_session
INSERT INTO hydra_oauth2_authentication_request
The inserts on these tables taking longer times.
The DB size is not that large also. We also ran the janitor script, It cleaned the request table but not the session table. We cleaned it manually.

We also integrated zipkin to check where it's taking time,
Will upgrading to hydra 1.10.7 will solve this issue? 
Are there any breaking changes I can expect, This will help us a lot while I'm analyzing changelog.

 
![zipkin time](https://user-images.githubusercontent.com/9514279/142882220-8c40f380-c57a-4a69-ab1b-e8558f2597a0.png)

**Comment by vinckr** [comment, marked as answer]
Hello @azajkhan 

Apologies for the late answer, this fell off over the holidays!

Hard to say what the issue is without having more insight into your infrastructure. From the Ory Hydra side this should definitely not be the case. 
Check the changelog here, there are no breaking changes in the versions above:
https://github.com/ory/hydra/blob/master/CHANGELOG.md#1103-2021-07-14

If possible I suggest you wait for the 1.11 release, it should not take too long anymore as the [milestone](https://github.com/ory/hydra/milestone/39) seems completed.

Maybe @grantzvolsky can chime in here quickly, as he is working on Hydra DB performance as well.

**Reply by grantzvolsky** [reply]
@azajkhan I would greatly appreciate if you wrote a test case that triggers the issue. Unfortunately there is little I can do without the ability to reproduce the problem.

**Reply by vinckr** [reply]
Hey @azajkhan  , just checking in if your problem still persists. 
I will mark this as answered for now, but feel free to come back to the discussion any time.
Thanks a ton :bee:
